src/adapter/src/coord/introspection.rs

```diff
@@ -476,11 +476,28 @@ impl SubscribeSpec {
     }
 }
 
-const SUBSCRIBES: &[SubscribeSpec] = &[SubscribeSpec {
-    introspection_type: IntrospectionType::ComputeErrorCounts,
-    sql: "SUBSCRIBE (
-        SELECT export_id, sum(count)
-        FROM mz_introspection.mz_compute_error_counts_raw
-        GROUP BY export_id
-    )",
-}];
+const SUBSCRIBES: &[SubscribeSpec] = &[
+    SubscribeSpec {
+        introspection_type: IntrospectionType::ComputeErrorCounts,
+        sql: "SUBSCRIBE (
+            SELECT export_id, sum(count)
+            FROM mz_introspection.mz_compute_error_counts_raw
+            GROUP BY export_id
+        )",
+    },
+    SubscribeSpec {
+        introspection_type: IntrospectionType::ComputeHydrationTimes,
+        sql: "SUBSCRIBE (
+            SELECT
+                export_id,
+                CASE count(*) = count(time_ns)
+                    WHEN true THEN max(time_ns)
+                    ELSE NULL
+                END AS time_ns
+            FROM mz_introspection.mz_compute_hydration_times_per_worker
+            WHERE export_id NOT LIKE 't%'
+            GROUP BY export_id
+            OPTIONS (AGGREGATE INPUT GROUP SIZE = 1)
+        )",
```

Comment:
I'd love to see some performance data around how expensive it is to maintain this subscribe. It seem to me it shouldn't be, but let's confirm that our intuition is correct. I think it would make sense to have a cluster with many indexes/MVs and see how this subscribe needs to spend effort to maintain its output.

Reply:
One datapoint is that there is no regression in the feature benchmarks. Of course the feature benchmarks are probably not targeting the scenario you have in mind and they only detect >10% regressions. I can run a separate test in staging.

Note that we could get rid of the two counts and perhaps make the query execution more efficient by using `mz_all` here:
```
SELECT
    export_id,
    CASE mz_unsafe.mz_all(time_ns IS NOT NULL)
        WHEN true THEN max(time_ns)
        ELSE NULL
    END AS time_ns
FROM mz_introspection.mz_compute_hydration_times_per_worker
WHERE export_id NOT LIKE 't%'
GROUP BY export_id
```

That would require introspection subscribes to use `mz_unsafe` functions then which I'm not quite sure how to wire up. Also not sure if it really would be more efficient.

Reply:
I tested in staging with:
* a size `50cc` cluster
* TPCH loadgen, SF 1, tick interval 1s
* full pk and fk indexes
* MVs on all 22 queries
* indexes on all 22 MVs

According to `mz_scheduling_elapsed`, the new subscribe dataflow had run for 690ms after half an hour. Given that a `50cc` cluster has a single CPU, this implies an average of ~0.04% CPU usage by the introspection subscribe.

Our Grafana metrics for transient dataflows show 0.05% in steady state and spikes up to 0.57%. But they included the existing introspection subscribe, as well as `SELECT`s running against that replica, so these are just upper boundaries on the CPU utilization of the new subscribe.